### PR TITLE
Domains: Add domains management list breadcrumbs

### PR DESCRIPTION
**Pull request by leonardost**
### Changes proposed in this Pull Request

This PR adds the breadcrumbs section for the new domain management list page. This is part of the domain management pages redesign described in pcYYhz-m2-p2.

Desktop view screenshot:

Annotation on 2021-10-15 at 19-54-43

Mobile view screenshot:

Annotation on 2021-10-15 at 19-55-36

### Testing instructions

Access the live Calypso link, go to "Upgrades > Domains", and append this feature flag to your browser's URL: `?flags=domains/management-list-redesign`. Ensure the page contains the breadcrumbs as shown in the screenshot above and that all links work as expected. The page still contains only the breadcrumbs.

This PR depends on #57038.

### CODE REVIEW

**Comment by poligilad-auto** [comment]
Hey @leonardost, cool cool.

1. There are some things I still have to finaliize about the button sizes in the header area. So I will get back to that later on.
2. I want us to start using the [icon set](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/?node-id=8%3A13) used in Gutenberg. I don't know how the implementation works, but the blogger experience team used some of those icons, @delputnam maybe you could chat with them to figure out how to do this? So the idea would be to replace all the current icons with these.
3. As to width for the area, I was thinking this would have the same max width as the reset of the page and other pages in WPcom: 1040px.
image
4. Should this PR also include this filter?
image
5. In mobile view the title should have the same styling, currently it's not bold enough.
6. "Add a domain" dropdown should also have icons.
7. In mobile view: for the dropdown menu of "Add a domain" and the "3 dots" we need a better solution so they stay in view. I guess they could both be at [position bottom left](https://wpcalypso.wordpress.com/devdocs/design/popover).
8. Also, this should be "sticky" in desktop view, I don't know how to test that at the moment.

**Comment by leonardost** [comment]
@poligilad-auto thank you for the review!

> 1. There are some things I still have to finaliize about the button sizes in the header area. So I will get back to that later on.

Ok!

> 2. I want us to start using the icon set used in Gutenberg. I don't know how the implementation works, but the blogger experience team used some of those icons, delputnam maybe you could chat with them to figure out how to do this? So the idea would be to replace all the current icons with these.

I'll take a look to see how we can use these icons.

> 3. As to width for the area, I was thinking this would have the same max width as the reset of the page and other pages in WPcom: 1040px.

Done.

> 4. Should this PR also include this filter?

That'll be part of another PR, as the filter is more complex to implement.

> 5. In mobile view the title should have the same styling, currently it's not bold enough.

Fixed.

> 6. "Add a domain" dropdown should also have icons.

Sorry, I had overlooked those. Added now.

> 7. In mobile view: for the dropdown menu of "Add a domain" and the "3 dots" we need a better solution so they stay in view. I guess they could both be at position bottom left.

Do they go out of view in your browser? They appear fine here:
Screen Shot 2021-10-20 at 15 57 30

Sometimes Chrome's positioning bugs when you're in desktop view and change to mobile view - when that happens, I have to refresh the page while in mobile view and then the popovers are positioned correctly.

> 8. Also, this should be "sticky" in desktop view, I don't know how to test that at the moment.

This is being a little tricky for some reason 🤔 I'll possibly address this in a future PR.

File: client/my-sites/domains/domain-management/list/site-domains.jsx [hunk 1]
```diff
@@ -185,6 +186,35 @@ export class SiteDomains extends Component {
 		);
 	}
 
+	renderBreadcrumbs() {
+		const { translate } = this.props;
+
+		const item = {
+			label: 'Domains',
```
Comment (rafaelgallani) [inline client/my-sites/domains/domain-management/list/site-domains.jsx]: There's a translation missing:

```suggestion
			label: translate( 'Domains' ),
```

**Review by rafaelgallani — CHANGES_REQUESTED** [review]
LGTM. Nice job 😄 

I left a single minor comment, once it's addressed it should be good to go code-wise.
There's also a minor visual bug I noticed:

image

Simulating the `Mobile S` viewport - in the `pt-br` language: 
The elements' width exceeds the viewport's width, so they get positioned like this. 
While I don't know what should be the best way to position them in this case, I think this should be addressed before merging this PR.

**Comment by poligilad-auto** [comment]
Hey @leonardost @rafaelgalani @delputnam @hambai, to address this issue, make this more suitable for mobile and improve design consistency for this I propose we do something like this:

image

So on desktop we use the ['small buttons'](https://wpcalypso.wordpress.com/devdocs/design/button) and for the mobile version we use the ['large buttons'](https://wpcalypso.wordpress.com/devdocs/design/button) but only with the icons. We use a 'fill button' for 'add a domain' and a button with no outline for the 'other' options.

**Comment by leonardost** [comment]
These are great suggestions @poligilad-auto, I've updated the mobile view for the buttons in accordance to them.

@rafaelgalani these should solve the layout problems you noticed:

Screen Shot 2021-10-26 at 14 56 53

**Comment by poligilad-auto** [comment]
Cool @leonardost.
1. I think you are still using Gridicons for the buttons.
3. I will give some final comments on all the instances of the header breadcrumbs later on next week to finalize this, but for now one comment is that it should always be the same height, weather it has buttons or has only breadcrumbs, currently the height changes. 
2. Regarding the '+' the idea was to change it for desktop view as well. So it would be "+ Add a domain".

image

**Pull request merged**